src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -98,6 +98,13 @@ static bool IsCompatible(Type t, Type source)
 
         static bool IsTypeATimeoutHandledByAnySaga(Type type, IEnumerable<Type> sagas)
         {
+            // MakeGenericType() throws an exception if passed a ref struct type
+            // Messages cannot be ref struct types
+            if (type.CustomAttributes.Any(a => a.AttributeType.Name == "IsByRefLikeAttribute"))
```

Comment:
[The attribute requires .net471 ](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.isbyreflikeattribute?view=net-8.0)and up so we check the string instead.

Reply:
> will be executed for each saga on startup

I'm not sure that's right.

This code will be executed once for each type that we pass to conventions. That will be [once for every scanned type](https://github.com/Particular/NServiceBus/blob/release-7.8/src/NServiceBus.Core/EndpointCreator.cs#L106) and [once for every type in a route assembly](https://github.com/Particular/NServiceBus/blob/release-7.8/src/NServiceBus.Core/Routing/AssemblyRouteSource.cs#L23).

Reply:
>This code will be executed once for each type that we pass to conventions. That will be [once for every scanned type](https://github.com/Particular/NServiceBus/blob/release-7.8/src/NServiceBus.Core/EndpointCreator.cs#L106) and [once for every type in a route assembly](https://github.com/Particular/NServiceBus/blob/release-7.8/src/NServiceBus.Core/Routing/AssemblyRouteSource.cs#L23).

Reply:
Then we should probably switch to the alternative I suggested

if (type.GetCustomAttributesData().Any(attr => attr.AttributeType.FullName == "System.Runtime.CompilerServices.IsByRefLikeAttribute"))

that should in theory be faster and have less side effects

Might be worth comparing the two

Reply:
I was misreading this part

https://github.com/Particular/NServiceBus/blob/216e3b18fc9c7feec1115fbfb467f455785edbbd/src/NServiceBus.Core/Sagas/Sagas.cs#L23-L27

ignoring that we only add the func to the convention

Reply:
> GetCustomAttributesData

[Maybe the additional overhead of creating the CustomAttributeData type might actually make GetCustomAttributeData slower](https://stackoverflow.com/questions/47184352/getcustomattributes-vs-getcustomattributesdata). Interesting